src/foam/nanos/auth/ThemeAuthService.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'ThemeAuthService',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+  flags: ['java'],
+
+  documentation: `Restrict login to the url that matches the spid of the user.`,
+
+  javaImports: [
+    'foam.nanos.auth.AuthenticationException',
+    'foam.nanos.auth.AuthService',
+    'foam.nanos.auth.User',
+    'foam.util.Password',
+    'foam.nanos.theme.Theme',
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'superSpid',
+      documentation: 'Set spid of a theme that is accessible to all users',
+      value: "",
+    }
+  ],
+
+  methods: [
+    {
+      name: 'login',
+      javaCode: `
+        return login_(x, identifier, password);
+      `
+    },
+    {
+      name: 'login_',
```

Comment:
I dont think you should override this method. Remove "login_" and move the code to "login"

Reply:
Thank you, I fixed this.